### DISCUSSION TITLE
Suddenly getting Error: app 'myapp' not deployed!

### DISCUSSION THREAD

**Discussion by mariusa** [piku/piku#299] [Q&A]
Hi,

After running few times `./piku.py restart myapp`, the app didn't start anymore. 

Last log is
```
web.1    | Tue Apr  4 12:41:34 2023 - received message 0 from emperor
web.1    | SIGINT/SIGTERM received...killing workers...
web.1    | [uwsgi-daemons] stopping daemon (pid: 33326): npm start 2>&1 | cat
web.1    | goodbye to uWSGI.
```

1. I don't why it didn't start afterwards

To solve this manually, I tried runing `./piku.py restart myapp`, but it gives
```
 Error: app 'myapp' not deployed!
```

The 2 `.ini` files exist in `uwsgi-available`, but folder `uwsgi-enabled` is empty. I couldn't figure out the difference between these 2 by looking on code.

```bash
$ piku ps
web:1
worker:1
```

2. What would be the cause & solution for this situation?

Thanks

**Comment by jfmatth** [comment]
I find that deploying the app again seems to help (sometimes)

**Comment by jfmatth** [comment]
> Hi,
> 
> After running few times `./piku.py restart myapp`, the app didn't start anymore.
> 
> Last log is
> 
> ```
> web.1    | Tue Apr  4 12:41:34 2023 - received message 0 from emperor
> web.1    | SIGINT/SIGTERM received...killing workers...
> web.1    | [uwsgi-daemons] stopping daemon (pid: 33326): npm start 2>&1 | cat
> web.1    | goodbye to uWSGI.
> ```
> 
> 1. I don't why it didn't start afterwards
> 
> To solve this manually, I tried runing `./piku.py restart myapp`, but it gives
> 
> ```
>  Error: app 'myapp' not deployed!
> ```
> 
> The 2 `.ini` files exist in `uwsgi-available`, but folder `uwsgi-enabled` is empty. I couldn't figure out the difference between these 2 by looking on code.
> 
> ```shell
> $ piku ps
> web:1
> worker:1
> ```
> 
> 2. What would be the cause & solution for this situation?
> 
> Thanks

Did you make any progress on this?

I did found a bug where uwsgi wasn't set to enabled, but started, so on a reboot of the machine, it wouldn't be started.  The piku interface doesn't show the status of the services, so that might be your issue.

hth

**Comment by chr15m** [comment]
@rcarmo perhaps we should change `restart` so that if it detects the app is not deployed it does a `deploy` instead of a restart. :thinking:

**Reply by rcarmo** [reply]
I’m Ok with that.
